plugin/cache/cache.go

```diff
@@ -228,7 +241,9 @@ func (w *ResponseWriter) Write(buf []byte) (int, error) {
 
 const (
 	maxTTL  = dnsutil.MaximumDefaulTTL
+	minTTL  = 0
 	maxNTTL = dnsutil.MaximumDefaulTTL / 2
+	minNTTL = 0
```

Comment:
in pkg/dnsutil/ttl.go we say the minimum ttl coredns allows is 5s, which may not be (currently) true, but certainly clashes with these values.

Reply:
Set minTTL and minNTTL to dnsutil.MinimalDefaultTTL.

Note that this required changing TestCacheZeroTTL as now zero ttl response is cached since